lib/common/socket/uv-binding.c.h

```diff
@@ -175,22 +241,43 @@ h2o_socket_t *do_import(h2o_loop_t *loop, h2o_socket_export_t *info)
 
     if (sock == NULL)
         return NULL;
-    if (uv_tcp_open((uv_tcp_t *)sock->uv.stream, info->fd) != 0) {
+    if (uv_tcp_open((uv_tcp_t *)sock->handle, info->fd) != 0) {
         h2o_socket_close(&sock->super);
         return NULL;
     }
 
     return &sock->super;
 }
 
+h2o_socket_t *h2o_uv_poll_create(h2o_loop_t *loop, int fd, uv_close_cb close_cb)
+{
+    struct st_h2o_uv_socket_t *sock = h2o_mem_alloc(sizeof(*sock));
+    memset(sock, 0, sizeof(*sock));
+    h2o_buffer_init(&sock->super.input, &h2o_socket_buffer_prototype);
+
+    uv_poll_t *poll = h2o_mem_alloc(sizeof(*poll));
+    if (uv_poll_init(loop, poll, fd) != 0) {
+        goto Error;
+    }
+    sock->handle = (uv_handle_t *)poll;
+    sock->close_cb = close_cb;
+    sock->poll.events = 0;
+    sock->handle->data = sock;
+    return &sock->super;
+Error:
+    uv_close((uv_handle_t *)poll, (uv_close_cb)free);
+    free(sock);
+    return NULL;
+}
```

Comment:
Please do not copy-paste the initialization tasks from `h2o_uv_socket_create`. Instead, you should extract the logics in common to a separate function and call it from the two create functions.